### PR TITLE
Add BeaconInventoryMock

### PR DESCRIPTION
**Pull request by thelooter**
# Description
Adds a Mock for the BeaconInventory 

# Checklist
The following items should be checked before the pull request can be merged.
- [x] Unit tests added.
- [x] Code follows existing code format.

# Info on creating a pull request
- Make sure that unit tests are added which test the relevant changes.
- Make sure that the changes follow the existing code format.

# For maintainer
When a PR is approved, the maintainer should label this PR with `release/*`.

- If the PR fixes a bug in MockBukkit, update the patch version. (if the current version is `0.4.1`, the new version should be `0.4.2`)
- If the PR adds a new feature to MockBukkit, update minor version. (if the current version is `0.4.1`, the new version should be `0.5.0`)

Note that a PR that fixes an `UnimplementedOperationException` should be considered a new version and not a bugfix.

### CODE REVIEW

**Comment by thelooter** [comment]
Yeah, I might have forgotten to add Tests to that class :sweat_smile:

File: src/main/java/be/seeseemelk/mockbukkit/inventory/BeaconInventoryMock.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import org.bukkit.inventory.BeaconInventory;
+import org.bukkit.inventory.BrewerInventory;
+import org.bukkit.inventory.InventoryHolder;
+import org.bukkit.inventory.ItemStack;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class BeaconInventoryMock extends InventoryMock implements BeaconInventory
+{
+	@Nullable ItemStack item;
```
Comment (Insprill) [inline src/main/java/be/seeseemelk/mockbukkit/inventory/BeaconInventoryMock.java]:
```suggestion
{

	private @Nullable ItemStack item;

File: src/main/java/be/seeseemelk/mockbukkit/inventory/BeaconInventoryMock.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	}
+
+	@Override
+	public @NotNull BeaconInventoryMock getSnapshot()
+	{
+		BeaconInventoryMock inventory = new BeaconInventoryMock(getHolder());
+		inventory.setItem(getItem());
+		return inventory;
+	}
+}
```
Comment (Insprill) [inline src/main/java/be/seeseemelk/mockbukkit/inventory/BeaconInventoryMock.java]:
```suggestion
	}

}

File: src/test/java/be/seeseemelk/mockbukkit/inventory/BeaconInventoryMockTest.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+	{
+		assertNotNull(inventory.getSnapshot());
+
+		inventory.setItem(new ItemStack(Material.EMERALD));
+		assertNotEquals(inventory, inventory.getSnapshot());
+		assertEquals(inventory.getItem(), inventory.getSnapshot().getItem());
+	}
+
+
+}
```
Comment (Insprill) [inline src/test/java/be/seeseemelk/mockbukkit/inventory/BeaconInventoryMockTest.java]:
```suggestion
	}

}

File: src/test/java/be/seeseemelk/mockbukkit/inventory/BeaconInventoryMockTest.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+public class BeaconInventoryMockTest
+{
+
+	ServerMock server;
+	BeaconInventoryMock inventory;
```
Comment (Insprill) [inline src/test/java/be/seeseemelk/mockbukkit/inventory/BeaconInventoryMockTest.java]:
```suggestion
	private ServerMock server;
	private BeaconInventoryMock inventory;

**Review by Insprill — CHANGES_REQUESTED** [review]

**Review by Insprill — APPROVED** [review]
LGTM

**Pull request merged**